### PR TITLE
Added reference property

### PR DESCRIPTION
**Pull request by wilcoschoneveld**
usage:
```
p = kec.scope('Bike Project')
p.part('Bike').property('ReferenceProperty').value = p.part('Rear Wheel')
```

With the following behavior:
- Setting value to None will also return a value of None
- Setting value to a Part will also return a value of Part
- All other value types are invalid

### CODE REVIEW

**Comment by wilcoschoneveld** [comment]
@jberends before I fix and add tests, what do you think?

File: pykechain/models/property_reference.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        :return: Part or None
+        """
+        if not self._value:
+            return None
+
+        return self._client.part(pk=self._value['id'])
+
+    @value.setter
+    def value(self, value):
+        if value and not isinstance(value, Part):
```
Comment (wilcoschoneveld) [inline pykechain/models/property_reference.py]: I think a value of `0` will unintentionally pass this test

Comment (jberends) [inline pykechain/models/property_reference.py]: good find

**Comment by jberends** [comment]
regarding your initial comment --> good behaviour and desired. This way you can 'unset' the reference property aswell.

In the KEC2 roadmap there will be a multiple reference property (list/set of references) down the line. Just to be aware of.

I fixed the tests, with your permission we can close and merge. Don't forget to update the CHANGELOG.rst before or after the merge.

File: pykechain/models/property_reference.py [hunk 2]
```diff
@@ -17,9 +17,11 @@ def value(self):
 
     @value.setter
     def value(self, value):
-        if value and not isinstance(value, Part):
-            raise ValueError("Reference must be a Part")
-
-        part_id = value.id if value else None
+        if isinstance(value, Part):
+            part_id = value.id
+        elif isinstance(value, (type(None), str)):
```
Comment (jberends) [inline pykechain/models/property_reference.py]: It seems that this line if not py27 compatible. We get a unicode back from the response, so we need to improve this check.

**Pull request merged**